app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomAdapter.kt

```diff
@@ -50,25 +59,83 @@ class ChatRoomAdapter(private val serverUrl: String) : RecyclerView.Adapter<Chat
         notifyItemInserted(0)
     }
 
-    fun updateItem(index: Int, message: MessageViewModel) {
-        dataSet[index] = message
-        notifyItemChanged(index)
+    fun updateItem(message: MessageViewModel) {
+        val index = dataSet.indexOfFirst { it.id == message.id }
```

Comment:
We can add an extension here that will be useful on other places.
```kotlin
public inline fun <T> List<T>.firstWithIndex(predicate: (T) -> Boolean): Pair<Int, T>? {
    val index = this.indexOfFirst(predicate)
    val data = if (index != -1) this[index] else null

    return if (data != null) Pair(index, data) else null
}

    fun updateItem(message: MessageViewModel) {
        dataSet.firstWithIndex { it.id == message.id }?.let { (index, _) ->
            dataSet[index] = message
            notifyItemChanged(index)
        }
    }
```

Reply:
I see what you mean but I think it makes it less clear.